File: JsonParser.java
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.File;
import java.io.IOException;

public class JsonParser {
    public static void main(String[] args) throws IOException {
        ObjectMapper mapper = new ObjectMapper();
        JsonNode rootNode = mapper.readTree(new File("SampleTestCase.json"));

        // Parse the "keys" object
        JsonNode keysNode = rootNode.get("keys");
        int n = keysNode.get("n").asInt();
        int k = keysNode.get("k").asInt();
        System.out.println("n: " + n);
        System.out.println("k: " + k);

        // Parse the roots
        for (JsonNode root : rootNode) {
            if (!root.get("keys").equals(rootNode.get("keys"))) {
                int x = Integer.parseInt(root.asText());
                JsonNode rootObject = rootNode.get(root.asText());
                String base = rootObject.get("base").asText();
                String value = rootObject.get("value").asText();
                int y = decodeValue(base, value);
                System.out.println("Root: x = " + x + ", y = " + y);
            }
        }
    }

    private static int decodeValue(String base, String value) {
        int y = 0;
        for (char c : value.toCharArray()) {
            y = y * Integer.parseInt(base) + Integer.parseInt(String.valueOf(c), Integer.parseInt(base));
        }
        return y;
    }
}

File: jsonparser.js
// json-parser.js

// Load the JSON file
const fs = require('fs');
const filePath = 'SampleTestCase.json';
const fileContent = fs.readFileSync(filePath, 'utf8');

// Parse the JSON file
const jsonData = JSON.parse(fileContent);

// Extract the keys object
const keys = jsonData.keys;
const n = keys.n;
const k = keys.k;

console.log(`n: ${n}, k: ${k}`);

// Extract the roots
const roots = {};
Object.keys(jsonData).forEach((key) => {
  if (key !== 'keys') {
    const root = jsonData[key];
    const base = parseInt(root.base, 10);
    const value = root.value;
    const decodedValue = parseInt(value, base);
    roots[key] = decodedValue;
  }
});

console.log('Roots:');
console.log(roots);
